src/Pages/UserPages/Explore/Explore.tsx

```diff
@@ -1,3 +1,262 @@
+import {
+  Box,
+  ButtonBase,
+  Chip,
+  Grid2 as Grid,
+  InputLabel,
+  MenuItem,
+  Pagination,
+  Select,
+  Skeleton,
+  Stack,
+  Tooltip,
+  Typography,
+} from "@mui/material";
+
+import { useLocation } from "react-router-dom";
+import { motion } from "framer-motion";
+
+import { PhotoCard } from "../../../Components/AdminSharedComponents/PhotoCard/PhotoCard";
+import { AxiosError } from "axios";
+import { useState, ReactNode, useEffect } from "react";
+import toast from "react-hot-toast";
+import {
+  apiClient,
+  getRoomDetails,
+  PORTAL_URLS,
+} from "../../../Api/END_POINTS";
+import BasicBreadcrumbs from "../../../Components/UserSharedComponents/BasicBreadcrumbs/BasicBreadcrumbs";
+
+interface FavoriType {
+  rooms: roomType;
+}
+
+interface FavoriResponseType {
+  data: {
+    favoriteRooms: FavoriType[];
+  };
+}
+interface roomType {
+  _id: string;
+  roomNumber: string;
+  images: string[];
+  price: number;
+}
+interface PickerData {
+  state: { data: { rooms: roomType[]; totalCount: number } };
+}
+interface AllRoomsResponseType {
+  data: {
+    rooms: roomType[];
+    totalCount: number;
+  };
+}
 export default function Explore() {
-  return <div>Explore</div>;
+  const location: PickerData = useLocation();
+  console.log(location);
+  const [favoriteRooms, setFavoriteRooms] = useState<FavoriType[]>([]);
+  const [rooms, SetRooms] = useState<roomType[]>(
+    location.state?.data?.rooms || []
+  );
+  const [totalCount, setTotalCount] = useState<number>(
+    location.state?.data?.totalCount || 1
+  );
+  const [page,setPage] = useState<number>(1);
+  const [size,setSize] = useState<number>(8);
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [selectedFilter, setSelectedFilter] = useState<
+    "highest" | "lowest" | null
+  >(null);
+
+  const getAllRooms = async (page:number , size:number) => {
+    try {
+      const response = await apiClient.get<AllRoomsResponseType>(
+        getRoomDetails,{
+          params:{page:page , size: size}
```

Comment:
We can use the shorthand syntax here as the name of the property is the same as the value.